pkg/3scale/amp/operator/highavailability_options_provider.go

```diff
@@ -296,3 +368,33 @@ func defaultBackendQueuesSentinelHosts() string {
 func defaultBackendQueuesSentinelRole() string {
 	return ""
 }
+
+func (h *HighAvailabilityOptionsProvider) validateRedisTLSFields(secretName string, fields []SecretField) []error {
+	var errs []error
+	for _, field := range fields {
+		val, err := h.secretSource.RequiredFieldValueFromRequiredSecret(secretName, field.secretFieldName)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("%w", err))
+		}
+		*field.field = val
+	}
+	return errs
+}
+
+func (h *HighAvailabilityOptionsProvider) validateRedisURLPrefix(redisUrl string) error {
+	if redisUrl == "" {
+		return fmt.Errorf("Redis URL cannot be empty in secret")
+	}
+	// If TLS is enabled, URL should start with "rediss://"
+	if h.apimanager.IsSystemRedisTLSEnabled() {
+		if !strings.HasPrefix(redisUrl, "rediss://") {
+			return fmt.Errorf("invalid URL, when TLS is enabled, URL must start with 'rediss://'")
```

Comment:
maybe just update the message, odds are if the prefix is wrong then the port is also. 
```suggestion
			return fmt.Errorf("invalid URL, when TLS is enabled, URL must start with 'rediss://, also confirm your port matches the TLS port set in your redis.conf'")
```

Reply:
Done.  Also errors accumulation added  for backend.
Regarding Port - thanks for this suggestion, as we discussed it, and it looks best what we can do without analysing redis.conf file.
Thank you